trinity/tx_pool/base.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from typing import (
+    cast,
+    List
+)
+
+from p2p.eth import (
+    Transactions
+)
+from p2p.peer import (
+    BasePeer,
+    ETHPeer,
+    PeerPool,
+    PeerPoolSubscriber,
+)
+from p2p.rlp import (
+    P2PTransaction
+)
+from p2p.service import (
+    BaseService
+)
+
+
+class TxPool(BaseService, PeerPoolSubscriber):
+    """
+    Minimal viable Transaction Pool
+    """
+    logger = logging.getLogger("trinity.tx_pool.TxPool")
+
+    def __init__(self, peer_pool: PeerPool) -> None:
+        super().__init__()
+        self.peer_pool = peer_pool
+
+    def register_peer(self, peer: BasePeer) -> None:
+        pass
+
+    async def _run(self) -> None:
+        self.logger.info("Running Tx Pool")
+
+        with self.subscribe(self.peer_pool):
+            while True:
+                peer: ETHPeer
+                peer, cmd, msg = await self.wait(
+                    self.msg_queue.get(), token=self.cancel_token)
+
+                if isinstance(cmd, Transactions):
+                    await self._handle_tx(peer, msg)
+
+    async def _handle_tx(self, peer: ETHPeer, msg: List[P2PTransaction]) -> None:
+        self.logger.info("Wohoo, let's do work!")
+        self.logger.info(str(msg))
+
+        for receiving_peer in self.peer_pool.peers:
+            receiving_peer = cast(ETHPeer, receiving_peer)
+            if receiving_peer is not peer:
+                receiving_peer.sub_proto.send_transactions(
+                    self._filter_tx_for_peer(receiving_peer, msg)
```

Comment:
The filtering should probably happen first, and then only send if the result of filtering is not an empty list.

Reply:
Yes, good catch. I plan to cover all of that with test as well.